webserver/server/app/services/DexSwap.ts

```diff
@@ -0,0 +1,41 @@
+import type { DexSwapResponse } from '../../../shared/models/DexSwap';
+import type { Asset } from '../../../shared/models/common';
+import type { PoolClient } from 'pg';
+import type { TransactionPaginationType } from './PaginationService';
+import type { Dex } from '../../../shared/models/common';
+import { sqlDexSwap } from '../models/dex/sqlDexSwap.queries';
+import { parseAssetItem, serializeAsset, valueToDex, dexToValue} from './utils';
+import { Direction } from '../../../shared/models/common';
+
+export async function dexSwap(
+  request: TransactionPaginationType & {
+    dbTx: PoolClient;
+    dexes: Array<Dex>;
+    assetPairs: {asset1: Asset, asset2: Asset}[];
+    limit: number;
+  }
+): Promise<DexSwapResponse> {
+  if (request.assetPairs.length === 0) return { swap: [] };
+
+  const swap = await sqlDexSwap.run({
+    after_tx_id: (request.after?.tx_id ?? -1)?.toString(),
+    until_tx_id: request.until.tx_id.toString(),
+    dexes: request.dexes.map(dex => dexToValue(dex)),
+    policy_id1: request.assetPairs.map(pair => parseAssetItem(pair.asset1?.policyId)),
+    asset_name1: request.assetPairs.map(pair => parseAssetItem(pair.asset1?.assetName)),
+    policy_id2: request.assetPairs.map(pair => parseAssetItem(pair.asset2?.policyId)),
+    asset_name2: request.assetPairs.map(pair => parseAssetItem(pair.asset2?.assetName)),
+    limit: request.limit.toString(),
+  }, request.dbTx);
+  return {
+    swap: swap.map(result => ({
+      tx_hash: result.tx_hash.toString('hex'),
+      dex: valueToDex(result.dex || '-1'),
```

Comment:
be careful with using `||` because for example, `0` is falsy in javascript. If you're doing a null-check, it's better to use `??`

Reply:
It is only because views return `type or null` value.
There is no option that the value will be really null.